packages/editor/src/components/post-preview-button/index.js

```diff
@@ -186,7 +188,12 @@ export default compose( [
 			isEditedPostNew,
 			isEditedPostSaveable,
 			isEditedPostAutosaveable,
+			isAutosavingPost,
 		} = select( 'core/editor' );
+		const {
+			hasMetaBoxes,
+			isSavingMetaBoxes,
+		} = select( 'core/edit-post' );
```

Comment:
> I don't know what this PR is trying to do exactly 

@youknowriad 
This PR is attempting to save metaboxes when a preview is triggered. That is, when the preview button is clicked, we save meta before displaying the preview so updated meta values will be included.

Reply:
> I think you probably want to tweak this condition instead 

Yes thanks for the suggestion... I looked at that originally without much luck because the previewing "condition" is not being tracked. It seemed more straightforward to add metabox saving as an effect of clicking the preview button only and code wise it is maybe simpler.  

to move the logic into `SET_META_BOXES_PER_LOCATIONS` in effects, i think we need to track `isPreviewing` to know this is a preview autosave not a regular autosave - we don't want to save metabox meta for regular autosaves, only when the user clicks preview - does that sound right? or should i instead try to add a variable to the autosave call, eg autosave( true ) would be a preview autosave?.

 > (Curious about the impact on revisions of all these save requests though)

no revisions are stored for meta currently in core - the meta is saved directly to the post (unfortunately). only one autosave revision is saved per post (per user).

Reply:
Why the preview autosave is not just a regular autosave? Is there anything different in the backend or is this the actual different, the need to not trigger ant autosave but an actual save including metaboxes? 

- Could we just trigger a regular save (instead of autosave) when previewing drafts?
- What should happen if we preview a published post and preview is clicked, should we save the metaboxes too? it seems like an unwanted changed could be published inadvertently?
 - If it's only for drafts, then I'd be fine adding a flag tracking that we're previwing.

Reply:
> Why the preview autosave is not just a regular autosave? 

to match core behavior: 

regular autosaves do not include saving post meta - they only save the title, excerpt and content. previews also save all meta so that previews can include changes made in meta fields which may affect how the post displays. This fixes the issue raised in https://github.com/WordPress/gutenberg/issues/11280. it also re-introduces the issue raised in https://core.trac.wordpress.org/ticket/20299.

I would love to work on a more complete solution - - perhaps as part of phase 2 - that would add meta revisioning to the autosave, so meta values would be associated with the autosave instead of the parent post, which would fix the unwanted saving of meta values to published posts when previewing. The code to achieve this is already available in a patch on the ticket and relies on my [Post Meta Revisioning](https://github.com/adamsilverstein/wp-post-meta-revisions) plugin.

> Could we just trigger a regular save (instead of autosave) when previewing drafts?

Yes, probably - however that might require more refactoring of how the preview loads its data?

> What should happen if we preview a published post and preview is clicked, should we save the metaboxes too? it seems like an unwanted changed could be published inadvertently?

Absolutely right - this is a tricky situation, if we save the metaboxes unwanted changes could be saved, if we don't save the metaboxes we risk the preview not showing the correct information. That is why adding meta revisioning and using that for the preview makes so much sense - it solves both problems.  Short of that, we probably want to match the _existing core behavior_ despite the possibility of unwanted data being saved. Since this is such a long standing and well documented bug it has (sadly) become the expected state of affairs. 

> If it's only for drafts, then I'd be fine adding a flag tracking that we're previwing.

its for drafts and published posts, both need to save meta when previewing.

Reply:
> its for drafts and published posts, both need to save meta when previewing.

Gotcha, this means you're right that we need to distinguish the "preview save" from an "autosave" in the state to be able to trigger the metaboxes save in `edit-post`.